### PR TITLE
Optimize the up_mdelay usage

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- Change up_mdelay with >= 10ms with nxsig_usleep
- Change up_mdelay with < 10ms with up_udelay 

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

File: arch/arm/src/c5471/c5471_ethernet.c [hunk 1]
```diff
@@ -1892,7 +1892,7 @@ static int c5471_ifup(struct net_driver_s *dev)
 
   putreg32((getreg32(EIM_CTRL) | EIM_CTRL_ESM_EN), EIM_CTRL);      /* enable ESM */
   putreg32((getreg32(ENET0_MODE) | ENET_MODE_ENABLE), ENET0_MODE); /* enable ENET */
-  up_mdelay(100);
+  nxsig_usleep(100000);
```
Comment (pkarashchenko) [inline arch/arm/src/c5471/c5471_ethernet.c:1895]: The `up_mdelay` does not depend in scheduler configuration while `nxsig_usleep` does.

**Comment by davids5** [comment]
Were all the up_ delays changes scrutinized see it interrupts were off?

**Comment by xiaoxiang781216** [comment]
> Were all the up_ delays changes scrutinized see it interrupts were off?

Yes, we need review carefully, but the replacement is only for the sleep >= 10ms, I don't believe anyone expect to sleep so much time in the interrupt handle.

File: arch/arm/src/c5471/c5471_ethernet.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm/src/c5471/c5471_ethernet.c:1895]: What's the dependence you mean? the schedule tick?

Comment (pkarashchenko) [reply arch/arm/src/c5471/c5471_ethernet.c:1895]: Yes

File: arch/arm/src/c5471/c5471_ethernet.c [hunk 2]
```diff
@@ -2134,7 +2135,7 @@ static void c5471_eimreset (struct c5471_driver_s *priv)
 
   putreg32((CLKM_CTL_RST_EXT_RESET | CLKM_CTL_RST_LEAD_RESET),
            CLKM_CTL_RST);
-  up_mdelay(2);
+  up_udelay(2000);
```
Comment (pkarashchenko) [inline arch/arm/src/c5471/c5471_ethernet.c:2138]: @xiaoxiang781216 do you expect better precision with `up_udelay` compared to `up_mdelay`? Or what is the motivation for this change?

File: arch/arm/src/lpc43xx/lpc43_cgu.c [hunk 3]
```diff
@@ -654,7 +654,7 @@ void lpc43_clockconfig(void)
 
   /* Delay around 10 milliseconds */
 
-  up_mdelay(XTAL_DELAY);
+  nxsig_usleep(XTAL_DELAY);
```
Comment (pkarashchenko) [inline arch/arm/src/lpc43xx/lpc43_cgu.c:657]: It is written in comment that `Delay around 10 milliseconds` and function is `lpc43_clockconfig` that is called from `__start`. Are you sure that scheduler is already running at this time?

In general I think that we need to be more careful with changing things that are working.

I will examine more carefully all other places. The first impression that we are trying to fix things that are not broken and might break the working cases.

Comment (davids5) [reply arch/arm/src/lpc43xx/lpc43_cgu.c:657]: I agree 100% - If this PR is to come in, every change needs to be checked for where it is run from. i.e. what stage is the system in and if the delay was uninterruptable for HW timing requirements.

Comment (davids5) [reply arch/arm/src/lpc43xx/lpc43_cgu.c:658]: @xiaoxiang781216 [this](https://github.com/apache/incubator-nuttx/pull/5493/commits/2baf39c1e23f8e72cbb7fe836f6b9c6da3d9c6cf)  type of regression  is the sort of thing we need to prevent from sneaking in in this set of changes.

Comment (xiaoxiang781216) [reply arch/arm/src/lpc43xx/lpc43_cgu.c:658]: I will check each location one by one, and mark this PR is ready for review when I finish. Of course, it need more review from you.

File: arch/arm/src/c5471/c5471_ethernet.c (hunk 2 above)
Comment (xiaoxiang781216) [reply arch/arm/src/c5471/c5471_ethernet.c:2138]: It isn't good to busy loop more than 1ms, using up_udelay with value >= 1000 will give the reader a more clean alert.
